### ISSUE TITLE
Keep current existing connections while updating/upgrading the cherokee instance

### ISSUE THREAD

**Issue by danielniccoli** [cherokee/webserver#147]
_Original author: tah...@gmail.com (August 04, 2008 12:03:25)_

It would be a nice to have.

Currently if changes made through cherokee-admin are applied, the instance
of cherokee is reinitiated loosing the whole queue of active connections.

We could implement a mechanism to allow launching a new instance of
cherokee with new configuration settings while and older one kept running
in parallel without accepting new connections. It should die once the queue
has been consumed. This would guarantee zero downtime.

In cherokee-admin this should translate as having some sort of checkbox
that allowed to keep at will the existing connections. It has to be
optional since sometimes it will be interesting just to kill everything.

This mechanism would also allow us to perform binary upgrades on the fly.

_Original issue: http://code.google.com/p/cherokee/issues/detail?id=81_
